File: Christmas Play.py
# 
# Website   : SPOJ
# Author    : Aadesh Salecha
# Created   : September 2016
#
for i in range(input()):
	n, k = map(int ,str(raw_input()).split())
	students = map(int, str(raw_input()).split())
	students.sort()

	min_distance = students[-1] - students[0]
	# print k/2,len(students) - k/2
	for j in range(k/2,n - (k/2 - int(k % 2 == 0))):
		upper = students[(j+k/2) - int(k % 2 == 0)]
		lower = students[j-k/2]
		min_distance = min(min_distance, upper - lower)

	print min_distance

File: Christmas Play.cpp
// # 
// # Website   : SPOJ
// # Author    : Aadesh Salecha
// # Created   : September 2016
// #
#include <iostream>
#include <algorithm>
#include <vector>

using namespace std;

int main(void)
{
	int T, n, k, tmp, min_distance, upper, lower;
	cin >> T;

	while (T--)
	{
		cin >> n >> k;
		vector <int> students;
		for(int i = 0; i < n; i++)
		{
			cin >> tmp;
			students.push_back(tmp);
		}

		sort(students.begin(), students.end());
		min_distance = students[n-1] - students[0];
		for(int i = k/2; i < n - (k/2 - int(k % 2 == 0)); i++)
		{
			upper = students[i + k/2 - int(k % 2 == 0)];
			lower = students[i - k/2];
			min_distance = min(min_distance, upper - lower);
		}

		cout << min_distance << endl;
	}
	return 0;
}
